src/ML-POD/compute_podfit.h

```diff
@@ -0,0 +1,214 @@
+/***************************************************************************
+               CESMIX-MIT Project
+
+ Contributing authors: Ngoc-Cuong Nguyen (cuongng@mit.edu, exapde@gmail.com)
+ ***************************************************************************/
+
+#ifdef COMPUTE_CLASS
+// clang-format off
+ComputeStyle(podfit,CPODFIT);
+// clang-format on
+#else
+
+#ifndef LMP_COMPUTE_PODFIT_H
+#define LMP_COMPUTE_PODFIT_H
+
+#include "compute.h"
+#include "pod.h"
+
+namespace LAMMPS_NS {
+
+class CPODFIT : public Compute {
+private:
+
+  class NeighList *list;
+  std::vector<std::string> globVector(const std::string& pattern, std::vector<std::string> & files);
+
+  bool is_a_number(std::string line);
+
+public:
+  struct datastruct {
+    std::string file_format;
+    std::string file_extension;
+    std::string data_path;
+    std::vector<std::string> data_files;
+    std::vector<std::string> filenames;
+
+    std::vector<int> num_atom;
+    std::vector<int> num_atom_cumsum;
+    std::vector<int> num_atom_each_file;
+    std::vector<int> num_config;
+    std::vector<int> num_config_cumsum;
+    int num_atom_sum;
+    int num_atom_min;
+    int num_atom_max;
+    int num_config_sum;
+
+    double *lattice=NULL;
+    double *energy=NULL;
+    double *stress=NULL;
+    double *position=NULL;
+    double *force=NULL;
+    int *atomtype=NULL;
+
+    int training = 1;
+    int normalizeenergy = 1;
+    int training_analysis = 1;
+    int test_analysis = 1;
+    int training_calculation = 0;
+    int test_calculation = 0;
+    int randomize = 1;
+    double percentage = 1.0;
+
+    double fitting_weights[12] = {0.0, 0.0, 0.0, 1, 1, 0, 0, 1, 1, 1, 1, 0};
+
+    void copydatainfo(datastruct &data) {
+      data.data_path = data_path;
+      data.file_format = file_format;
+      data.file_extension = file_extension;
+      data.data_files = data_files;
+      data.filenames = filenames;
+      data.training_analysis = training_analysis;
+      data.test_analysis = test_analysis;
+      data.training_calculation = training_calculation;
+      data.test_calculation = test_calculation;
+      data.percentage = percentage;
+      data.randomize = randomize;
+      data.training = training;
+      data.normalizeenergy = normalizeenergy;
+      for (int i = 0; i < 12; i++)
+        data.fitting_weights[i] = fitting_weights[i];
+    }
+
+    void freememory(int backend)
+    {
+      TemplateFree(lattice, backend);
+      TemplateFree(energy, backend);
+      TemplateFree(stress, backend);
+      TemplateFree(position, backend);
+      TemplateFree(force, backend);
+      TemplateFree(atomtype, backend);
+    }
+  };
+
+  struct neighborstruct {
```

Comment:
Same as for the pair style, headers for compute styles should not be defining and implementing structs. Instead there should be a pointer to it and a forward declaration.